src/Provider.ts

```diff
@@ -546,16 +540,24 @@ export const Provider: React.FC = ({ children }) => {
     readPendingMapRef.current = new WeakMap()
   }
 
+  const [state, setStateOrig] = useState(initialState)
+  const lastStateRef = useRef<State | null>(null)
+  const setState = (setStateAction: SetStateAction<State>) => {
+    lastStateRef.current = null
+    setStateOrig(setStateAction)
+  }
+
+  useIsoLayoutEffect(() => {
+    if (state !== initialState) {
```

Comment:
It feels weird, because this only works for initial Provider mount.
Anyway, let's move forward. We have some tests so it won't break.